### DISCUSSION TITLE
MetroLog to .Net MAUI in a Simple Way

### DISCUSSION THREAD

**Discussion by mgungorchamp** [roubachof/MetroLog#25] [General]
I added these two functions

`    private static ILoggerFactory loggerFactory = LoggerFactory.Create(builder => builder.AddStreamingFileLogger(options =>
    {
        options.RetainDays = 2;               
        options.FolderPath = Path.Combine(FileSystem.CacheDirectory, "InfoBoardLogs");
    }));

    public static ILogger Logger(string categoryName) {
        return loggerFactory.CreateLogger(categoryName);
    }`

And used in each file I need logging like the following
`private readonly ILogger _logger;`

` _logger = Utilities.Logger(nameof("MyFile"));`   
or
`_logger = Utilities.Logger(nameof(FileDownloadService));`

and use anywhere like this

`_logger.LogInformation($"**Welcome");`

Youtube video showing the usage 
https://youtu.be/8CPXDHQlBa0

**Comment by roubachof** [comment]
You can already get a logger with a static method, so I don't really understand the point of this helper.

```private static readonly ILogger Log = LoggerFactory.GetLogger(nameof(DownloadService));```

**Reply by mgungorchamp** [reply]
To shorten the chain of calls, instead of Utilities. LoggerFactory.Create().CreateLogger(categoryName);

**Reply by roubachof** [reply]
This is a bad way to create loggers: it would mean you create a logger factory for each logger. The correct way has the same length of call that your helper. Look again at my answer.

**Reply by mgungorchamp** [reply]
It's static, and static is initialized only once. Am I missing something?

**Reply by roubachof** [reply]
I think you misunderstood, what I mean is that 
`Utilities.Logger(nameof(FileDownloadService))` has the same chain of calls than `LoggerFactory.GetLogger(nameof(DownloadService))`.
So I don't see the point of this helper.

**Reply by mgungorchamp** [reply]
Oh, I must have missed that function exist...  that's better.  Thank you.

**Comment by jargule** [comment]
Will it support .NET 9.0 or maybe .NET 8?
